jmespath-jackson-jr/src/main/java/io/burt/jmespath/jacksonjr/JacksonJrRuntime.java

```diff
@@ -0,0 +1,225 @@
+    private final JSON json;
+    public JacksonJrRuntime() {
+        this(RuntimeConfiguration.defaultConfiguration());
+    }
+
+    public JacksonJrRuntime(RuntimeConfiguration configuration) {
+        this(configuration, JSON.builder()
+                .treeCodec(new JacksonJrsTreeCodec())
+                .build());
+    }
+    public JacksonJrRuntime(RuntimeConfiguration configuration, JSON json) {
+        super(configuration);
+        this.json = json;
+    }
+
+    @Override
+    public TreeNode parseString(String str) {
+        try {
+            return json.treeFrom(str);
+        } catch (IOException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    private static class JrsArrayListWrapper extends AbstractList<TreeNode> {
+        private final JrsArray array;
+
+        JrsArrayListWrapper(JrsArray array) {
+            this.array = array;
+        }
+
+        @Override
+        public TreeNode get(int index) {
+            return array.get(index);
+        }
+
+        @Override
+        public int size() {
+            return array.size();
+        }
+    }
+
+    @Override
+    public List<TreeNode> toList(TreeNode value) {
+        if (value == null) {
+            return Collections.emptyList();
+        }
+        if (value.isArray()) {
+            return new JrsArrayListWrapper((JrsArray) value);
+        } else if (value.isObject()) {
+            JrsObject object = (JrsObject) value;
+            List<TreeNode> list = new ArrayList<>(object.size());
+            Iterator<Map.Entry<String, JrsValue>> iterator = object.fields();
+
+            while (iterator.hasNext()) {
+                Map.Entry<String, TreeNode> entry = (Map.Entry) iterator.next();
+                list.add(entry.getValue());
+            }
+            return list;
+        } else {
+            return Collections.emptyList();
+        }
+    }
+
+    @Override
+    public String toString(TreeNode value) {
+        if (value.asToken().equals(JsonToken.VALUE_STRING)) {
+            return ((JrsString) value).asText();
+        } else {
+            try {
+                return json.asString(value);
+            } catch (IOException e) {
+                return "";
+            }
+        }
+    }
+
+    @Override
+    public Number toNumber(TreeNode value) {
+        if (value.isValueNode() && ((JrsValue) value).isNumber()) {
+            JrsNumber number = (JrsNumber) value;
+            return number.getValue();
+        } else return null;
+    }
+
+    @Override
+    public boolean isTruthy(TreeNode value) {
+        // false, null, empty lists, empty objects, empty strings.
+        if (value.isContainerNode()) {
+            return value.size() > 0;
+        } else if (value.isValueNode()) {
+            if (value.asToken().equals(JsonToken.VALUE_STRING)) {
+                return !((JrsString) value).asText().isEmpty();
+            } else return !value.asToken().equals(JsonToken.VALUE_FALSE) &&
```

Comment:
Nitpick: please use block style for all conditionals – especially if the conditional spans multiple lines.

---

I would write lines 119 to 122 like this to avoid all the noise:

```java
switch (value.asToken()) {
  case Json.VALUE_STRING:
    return !value.asToken().asString().isEmpty(); // assuming JrsString#asText and JsonToken#asString are equivalent
  case JsonToken.VALUE_FALSE:
  case JsonToken.VALUE_NULL:
    return false;
  default:
    return true;
}
```

Maybe the whole method could fit in the `switch`, but I'm unsure if a token of type e.g. `START_ARRAY` can tell us if it represents an empty array. I don't know which token gets attached to the `TreeNode` object.

---

When I read this I also ask myself how `0` isn't falsy in JMESPath, but it doesn't seem to be.

Reply:
replaced with the switch, 0 is truthy...

the comment `// false, null, empty lists, empty objects, empty strings.` was from the `isTruthy` to help me implement, I removed it.